velox/expression/CastExpr-inl.h

```diff
@@ -301,6 +325,81 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  int precision = getDecimalPrecisionScale(*fromType).first;
+  int scale = getDecimalPrecisionScale(*fromType).second;
+  // A varchar's size is estimated with unscaled value digits, dot, leading
+  // zero, and possible minus sign.
+  int varcharSize = precision + !!(scale > 0) + !!(precision == scale) + 1;
+  // Calculate the max total size of the raw string buffer.
+  size_t maxTotalResultBytes = rows.countSelected() * varcharSize;
+
+  auto flatResult = result->asFlatVector<StringView>();
+  Buffer* buffer = flatResult->getBufferWithSpace(maxTotalResultBytes);
+  char* rawBuffer = buffer->asMutable<char>() + buffer->size();
+  buffer->setSize(buffer->size() + maxTotalResultBytes);
+
+  applyToSelectedNoThrowLocal(context, rows, result, [&](vector_size_t row) {
+    if (simpleInput->isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      auto unscaledValue = simpleInput->valueAt(row);
+      const char* startPosition = rawBuffer;
+      if (unscaledValue == 0) {
+        *rawBuffer = '0';
```

Comment:
Can combine L358 and L359 to *rawBuffer++ = `0`.

Reply:
Fixed.